build/build.mk

```diff
@@ -1,14 +1,40 @@
 .PHONY : build
 build:
-	@($(MAVEN_BIN) clean package -DskipTests -P \!deb,\!rpm,\!tarball,\!test)
+	@($(MAVEN_BIN) clean package -DskipTests -P \!deb,\!rpm,\!tarball,\!test,\!tarball)
 
 .PHONY : test
 test:
 	@($(MAVEN_BIN) clean test -P test)
 
 CUR_DIR:=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
-GOMODULE_DIR:=$(CUR_DIR)/src/go/
+GOMODULE_DIR:=$(CUR_DIR)/gojmx/
 
 go-test: godeps build
 	@echo $(GOMODULE_DIR)
-	@cd $(GOMODULE_DIR); go clean -testcache; go test -v -timeout 300s github.com/newrelic/nrjmx
\ No newline at end of file
+	@cd $(GOMODULE_DIR); go clean -testcache; go test -v -timeout 300s github.com/newrelic/nrjmx/gojmx
+
+DOCKER_THRIFT=$(DOCKER_BIN) run --rm -t \
+					--name "nrjmx-code-generator" \
+					-v $(CURDIR):/src/nrjmx \
+					-w /src/nrjmx \
+					nrjmx-code-generator
+
+.PHONY : code-gen-deps
+code-gen-deps:
+		@($(DOCKER_BIN) build -t nrjmx-code-generator ./commons/.)
+
+.PHONY : code-gen-deps, code-gen
+code-gen: 
+	@($(DOCKER_THRIFT) thrift -r --out src/main/java/ --gen java ./commons/nrjmx.thrift)
+	@($(DOCKER_THRIFT) thrift -r --out gojmx/ --gen go:package_prefix=github.com/newrelic/nrjmx/gojmx/,package=nrprotocol ./commons/nrjmx.thrift)
+
```

Comment:
I tried `make code-gen` and failed. shouldn't this be:
```
.PHONY : code-gen-deps
code-gen-deps:
		@($(DOCKER_BIN) build -t nrjmx-code-generator ./commons/.)

.PHONY : code-gen
code-gen: code-gen-deps
	@($(DOCKER_THRIFT) thrift -r --out src/main/java/ --gen java ./commons/nrjmx.thrift)
	@($(DOCKER_THRIFT) thrift -r --out gojmx/ --gen go:package_prefix=github.com/newrelic/nrjmx/gojmx/,package=nrprotocol ./commons/nrjmx.thrift)
```

Reply:
tackling this in a new PR